src/components/Reports/GoalCalculator/GoalCalculatorTestWrapper.tsx

```diff
@@ -1,59 +1,264 @@
+import { InMemoryCache } from '@apollo/client';
 import { ThemeProvider } from '@emotion/react';
 import { MockLinkCallHandler } from 'graphql-ergonomock/dist/apollo/MockLink';
 import { SnackbarProvider } from 'notistack';
 import { DeepPartial } from 'ts-essentials';
 import TestRouter from '__tests__/util/TestRouter';
 import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
-import { PrimaryBudgetCategoryEnum } from 'src/graphql/types.generated';
+import { PrimaryBudgetCategoryEnum, SubBudgetCategoryEnum } from 'src/graphql/types.generated';
 import theme from 'src/theme';
 import { GoalCalculationQuery } from './Shared/GoalCalculation.generated';
 import { GoalCalculatorProvider } from './Shared/GoalCalculatorContext';
+import { 
+  CreateSubBudgetCategoryMutation,
+  DeleteSubBudgetCategoryMutation,
+  UpdatePrimaryBudgetCategoryMutation,
+  UpdateSubBudgetCategoryMutation,
+} from './SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.generated';
 
 interface GoalCalculatorTestWrapper {
   onCall?: MockLinkCallHandler;
   children?: React.ReactNode;
+  cache?: InMemoryCache;
+  useDynamicMocks?: boolean;
```

Comment:
Your dynamic mocks logic is very elaborate! However, the tests pass without `useDynamicMocks={true}`, so it doesn't seem to be necessary. Unless I'm missing something, can you remove the logic related to dynamic mocks?